helpdesk/management/commands/get_email.py

```diff
@@ -234,17 +234,17 @@ def process_queue(q, logger):
         logger.info("Found %s messages in local mailbox directory" % str(len(mail)))
         for m in mail:
             logger.info("Processing message %s" % str(m))
-            f = open(m, 'r')
-            ticket = ticket_from_message(message=f.read(), queue=q, logger=logger)
-            if ticket:
-                logger.info("Successfully processed message %s, ticket/comment created." % str(m))
-                try:
-                    # unlink(m) #delete message file if ticket was successful
-                    logger.info("Successfully deleted message %s." % str(m))
-                except:
-                    logger.error("Unable to delete message %s." % str(m))
-            else:
-                logger.warn("Message %s was not successfully processed, and will be left in local directory" % str(m))
+            with open(m, 'r') as f:
+                ticket = ticket_from_message(message=f.read(), queue=q, logger=logger)
+                if ticket:
+                    logger.info("Successfully processed message %s, ticket/comment created." % str(m))
+                    try:
+                        # unlink(m) #delete message file if ticket was successful
```

Comment:
I modeled this behavior after how JIRA takes in new tickets. New emails are dumped in a directory, helpdesk reads email to create a ticket, and then deletes it once confirmed successful to ensure that the same email doesn't generate duplicate tickets. I personally don't think there's a good way to avoid doing so. A compromise might be that emails could simply be renamed, rather than deleted, but that might not be desirable, it could fill up a drive with duplicate data. I would prefer simply deleting them; possibly documentation could be better improved to make it clear that files are permanently deleted and to be sure this is the behavior you want.